libraries/AP_Notify/ToshibaLED_I2C.cpp

```diff
@@ -37,35 +35,39 @@ extern const AP_HAL::HAL& hal;
 
 bool ToshibaLED_I2C::hw_init()
 {
-    // first look for led on external bus
-    _dev = std::move(hal.i2c_mgr->get_device(TOSHIBA_LED_I2C_BUS_EXTERNAL, TOSHIBA_LED_I2C_ADDR));
-    if (!_dev || !_dev->get_semaphore()->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
-        return false;
-    }
+    bool ret = false;
+    bool enabled[ARRAY_SIZE(busnum)] {};
+    
+    for (uint8_t i=0; i<ARRAY_SIZE(busnum); i++) {
+        struct toshiba_dev &dev = device[i];
 
-    // enable the led
-    bool ret = _dev->write_register(TOSHIBA_LED_ENABLE, 0x03);
-
-    // on failure try the internal bus
-    if (!ret) {
-        // give back external bus semaphore
-        _dev->get_semaphore()->give();
-        // get internal I2C bus driver
-        _dev = std::move(hal.i2c_mgr->get_device(TOSHIBA_LED_I2C_BUS_INTERNAL, TOSHIBA_LED_I2C_ADDR));
-        if (!_dev || !_dev->get_semaphore()->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
-            return false;
+        dev.dev = std::move(hal.i2c_mgr->get_device(busnum[i], TOSHIBA_LED_I2C_ADDR));
+        if (!dev.dev || !dev.dev->get_semaphore()->take(HAL_SEMAPHORE_BLOCK_FOREVER)) {
+            continue;
         }
-        ret = _dev->write_register(TOSHIBA_LED_ENABLE, 0x03);
-    }
+        dev.dev->set_retries(10);
 
-    // update the red, green and blue values to zero
-    uint8_t val[4] = { TOSHIBA_LED_PWM0, _led_off, _led_off, _led_off };
-    ret &= _dev->transfer(val, sizeof(val), nullptr, 0);
+        // enable the led        
+        enabled[i] = dev.dev->write_register(TOSHIBA_LED_ENABLE, 0x03);
```

Comment:
I'm not sure how to fix this, but doesn't this go against our "rule" not to do writes to I2C devices without really knowing what is there? Maybe try first to read the ENABLE register and check it is 0 (the default value)?

I realize this was done before, so I'm not fussed about it, but would be nice to make it more robust.